### PR TITLE
Add support for accessing annotations from method invocations

### PR DESCRIPTION
**Pull request by lisachenko**

### CODE REVIEW

File: src/Go/Aop/Support/AnnotatedReflectionMethod.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * Go! OOP&AOP PHP framework
```
Comment (cordoval) [inline src/Go/Aop/Support/AnnotatedReflectionMethod.php:3]: same reintroduction of stuff

File: src/Go/Aop/Support/AnnotatedReflectionMethod.php [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    private static function getReader()
+    {
+        if (!self::$annotationReader) {
+            // TODO: ugly global dependecy, decide how to inject it more friendly
+            self::$annotationReader = AspectKernel::getInstance()->getContainer()->get('aspect.annotation.reader');
+        };
+
+        return self::$annotationReader;
+    }
+}
```
Comment (cordoval) [inline src/Go/Aop/Support/AnnotatedReflectionMethod.php:62]: use php-cs-fixer

File: src/Go/Core/GoAspectContainer.php [hunk 3]
```diff
@@ -71,9 +71,17 @@ public function __construct()
...
+        $this->share('aspect.annotation.reader', function ($container) {
+            $options = $container->get('kernel.options');
+            $reader  = new AnnotationReader();
+            if (!empty($options['cacheDir'])) {
+                $reader  = new FileCacheReader(
+                    $reader,
+                    $options['cacheDir'] . DIRECTORY_SEPARATOR . '_annotations' . DIRECTORY_SEPARATOR,
+                    $options['debug']
+                );
+            }
```
Comment (cordoval) [inline src/Go/Core/GoAspectContainer.php:83]: leave a line break before a return

File: src/Go/Aop/Support/AnnotatedReflectionMethod.php (hunk 1 above)
Comment (lisachenko) [inline src/Go/Aop/Support/AnnotatedReflectionMethod.php:3]: no, this patch was applied before changing the name, so version in master is correct: https://github.com/lisachenko/go-aop-php/blob/master/src/Go/Aop/Support/AnnotatedReflectionMethod.php. However, there are branches with new files that are not merged yet, I will take care.

File: src/Go/Aop/Support/AnnotatedReflectionMethod.php (hunk 2 above)
Comment (lisachenko) [inline src/Go/Aop/Support/AnnotatedReflectionMethod.php:62]: Thanks! I should start to use this tool at home too.
BTW, do you know how to enable scrutinizer to send cs-fix PRs?

Comment (cordoval) [inline src/Go/Aop/Support/AnnotatedReflectionMethod.php:62]: you can just do:

```
gush p:fix
```

for your pleasure in any Pr

Comment (lisachenko) [inline src/Go/Aop/Support/AnnotatedReflectionMethod.php:62]: great :thumbsup:

**Pull request merged**